.github/workflows/release.yml

```diff
@@ -5,6 +5,18 @@ on:
     types: created
 
 jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2
+      with:
+        node-version: '10.x'
+        registry-url: 'https://registry.npmjs.org'
+    - run: npm install
+    - run: npm publish
```

Comment:
Maybe you could use https://github.com/marketplace/actions/npm-publish
Your way should works though.